csharp/src/Ice/DefaultsAndOverrides.cs

```diff
@@ -98,16 +98,15 @@ internal DefaultsAndOverrides(Communicator communicator, ILogger logger)
             {
                 OverrideCompress = true;
                 OverrideCompressValue = communicator.GetPropertyAsInt("Ice.Override.Compress") > 0;
-                if (!BZip2.Supported() && OverrideCompressValue)
+                if (!BZip2.Supported && OverrideCompressValue)
                 {
-                    string lib = AssemblyUtil.IsWindows ? "bzip2.dll" : "libbz2.so.1";
-                    Console.Error.WriteLine("warning: " + lib + " not found, Ice.Override.Compress ignored.");
+                    Console.Error.WriteLine($"warning: {BZip2.LibName} not found, Ice.Override.Compress ignored.");
```

Comment:
I would change the message to just "compression not supported, Ice.Override.Compress ignored", we don't actually know here why it's not supported... probably because the library is not found but it could also be because of a library mismatch, etc.